src/test/java/org/simulator/sbml/SBMLTestSuiteTest.java

```diff
@@ -144,50 +147,77 @@ public void testModel() throws FileNotFoundException, IOException {
-                // initialize interpreter
-                SBMLinterpreter interpreter = null;
-                boolean exceptionInInterpreter = false;
-                try {
-                    interpreter = new SBMLinterpreter(model, 0, 0, 1,
-                            amountHash);
-                } catch (SBMLException e) {
-                    exceptionInInterpreter = true;
-                } catch (ModelOverdeterminedException e) {
-                    exceptionInInterpreter = true;
-                }
-                Assert.assertNotNull(interpreter);
-                Assert.assertFalse(exceptionInInterpreter);
-
                 // get timepoints
                 CSVImporter csvimporter = new CSVImporter();
                 MultiTable inputData = csvimporter.convert(model, csvfile);
                 double[] timepoints = inputData.getTimePoints();
                 duration = timepoints[timepoints.length - 1]
                         - timepoints[0];
 
-                if ((solver != null) && (interpreter != null)) {
-                    // System.out.println(sbmlFileType + " " + solver.getName());
-                    solver.setStepSize(duration / steps);
+                if (model.getExtension(CompConstants.shortLabel) == null){
+
+                    AbstractDESSolver solver = new RosenbrockSolver();
+                    // initialize interpreter
+                    SBMLinterpreter interpreter = null;
+                    boolean exceptionInInterpreter = false;
+                    try {
+                        interpreter = new SBMLinterpreter(model, 0, 0, 1,
+                                amountHash);
+                    } catch (Exception e) {
+                        exceptionInInterpreter = true;
+                    }
+                    Assert.assertNotNull(interpreter);
+                    Assert.assertFalse(exceptionInInterpreter);
+
+                    if ((solver != null) && (interpreter != null)) {
+
+                        solver.setStepSize(duration / steps);
+
+                        // solve
+                        MultiTable solution = null;
+                        boolean errorInSolve = false;
+                        try {
+                            solution = solver.solve(interpreter,
+                                    interpreter.getInitialValues(), timepoints);
+                        } catch (DerivativeException e) {
+                            errorInSolve = true;
+                        }
+                        Assert.assertNotNull(solution);
+                        Assert.assertFalse(errorInSolve);
+                        Assert.assertFalse(solver.isUnstable());
+
+                        // compute distance
+                        QualityMeasure distance = new RelativeEuclideanDistance();
+                        double dist = distance.distance(solution, inputData);
+                        Assert.assertTrue(dist <= 0.2);
```

Comment:
I'd recommend to always indicate the data type also for literal numbers, here, for instance, `0.2d` so that the compiler knows it is a `double` value. If you do it always, you will not forget it in situations where it can make a difference (e.g., when dividing integer numbers).

Reply:
@draeger, this has been removed from the code now. The distance has been now updated to the MaxDivergenceTolerance.